crowdnotifier-sdk/src/androidTest/java/org/crowdnotifier/android/sdk/MatchingTest.java

```diff
@@ -43,198 +49,321 @@ public void init() {
 
-		ArrayList<ProblematicEventInfo> problematicEvents = new ArrayList<>();
-		problematicEvents
-				.add(new ProblematicEventInfo(venuePrivateKey, time - 1 * 60 * 60 * 1000l, time - 10, encryptedMessage,
-						nonce, r2));
+		//Setup Health Authority
+		KeyPair haKeyPair = createHAKeyPair();
 
-		List<ExposureEvent> exposureEvents = CrowdNotifier.checkForMatches(problematicEvents, context);
+		//Setup Location Owner
+		byte[] notificationKey = new byte[Box.SECRETKEYBYTES];
+		sodium.crypto_secretbox_keygen(notificationKey);
 
-		assertEquals(1, exposureEvents.size());
-		assertEquals(time - 10, exposureEvents.get(0).getEndTime());
-		assertEquals(time - 1 * 60 * 60 * 1000l, exposureEvents.get(0).getStartTime());
-		assertEquals(message, exposureEvents.get(0).getMessage());
-	}
+		long qrCodeValidFrom = currentTime - ONE_DAY_IN_MILLIS;
+		long qrCodeValidTo = currentTime + ONE_DAY_IN_MILLIS;
+		Location location = new Location(haKeyPair.publicKey, Qr.QRCodeContent.VenueType.OTHER, "Name", "Location",
+				"Room", notificationKey, qrCodeValidFrom, qrCodeValidTo);
+		Backend.QRCodeTrace qrTrace = location.getQrCodeTrace();
+		Qr.QRCodeEntry qrEntry = location.getQrCodeEntry();
+
+		//User checks in with App
+		String urlPrefix = "https://test.com";
+		VenueInfo venueInfo =
+				CrowdNotifier.getVenueInfo(urlPrefix + "?v=2#" + Base64Util.toBase64(qrEntry.toByteArray()), urlPrefix);
 
+		long arrivalTime = currentTime - ONE_HOUR_IN_MILLIS;
+		long departureTime = currentTime + ONE_HOUR_IN_MILLIS;
+		CrowdNotifier.addCheckIn(arrivalTime, departureTime, venueInfo, context);
 
-	@Test
-	public void testMatchingEnteredBefore() {
-		long time = System.currentTimeMillis();
-		byte[] notificationKey = Base64Util.fromBase64("HYZf4ROIMIp12Jr521JS3fttAmV4y1vATkx3MhTFB-E");
+		//Venue Owner Creates PreTraces
+		long exposureStart = currentTime - ONE_HOUR_IN_MILLIS;
+		long exposureEnd = currentTime;
 		String message = "This is a message";
-		byte[] nonce = getRandomValue(Box.NONCEBYTES);
-		byte[] encryptedMessage = encryptMessage(notificationKey, message, nonce);
-		byte[] r1 = getRandomValue(Box.SECRETKEYBYTES);
-		byte[] r2 = getRandomValue(Box.SECRETKEYBYTES);
-		String s = "a";
-		long validFrom = 0;
-		long validTo = 1000;
-		byte[] infoBytes = getInfoBytes(s, s, s, notificationKey, Qr.QRCodeContent.VenueType.OTHER, validFrom, validTo);
-		byte[] venuePublicKey = new byte[Box.PUBLICKEYBYTES];
-		byte[] venuePrivateKey = new byte[Box.SECRETKEYBYTES];
-		int result = sodium.crypto_box_seed_keypair(venuePublicKey, venuePrivateKey, hashInfo(infoBytes, r1, r2));
-		if (result != 0) { throw new RuntimeException("crypto_box_seed_keypair returned a value != 0"); }
-		VenueInfo venueInfo =
-				new VenueInfo(s, s, s, Qr.QRCodeContent.VenueType.OTHER, venuePublicKey, notificationKey, r1, validFrom, validTo);
+		List<Backend.PreTraceWithProof> preTraceWithProofList =
+				createPreTrace(qrTrace, exposureStart, exposureEnd, notificationKey, message);
```

Comment:
As commented elsewhere, I don't think `message` should be set in `createPreTrace`

Reply:
Same as above: This is only temporary. Our current (debug) version allows the VenueOwner to set a message. We will remove this in the future.